### PR TITLE
APP-1630 Scenarios to start conversation

### PR DESCRIPTION
**Pull request by robson-sanchez**
- Configure certificate directory in the bootstrap
- Steps to authenticate user given a certificate
- Steps to start conversation

### CODE REVIEW

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/steps/TicketSteps.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import org.symphonyoss.symphony.clients.model.SymMessage;
+import org.symphonyoss.symphony.clients.model.SymStream;
+import org.symphonyoss.symphony.clients.model.SymUser;
+
+/**
+ * Class responsible for managing ticket steps.
+ * <p>
+ * Created by rsanchez on 23/02/18.
+ */
+@Component
```
Comment (crepache) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/steps/TicketSteps.java:19]: Is the annotation required?

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/steps/UserSteps.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+/**
+ * Class responsible for managing user steps.
+ * <p>
+ * Created by rsanchez on 23/02/18.
+ */
+@Component
```
Comment (crepache) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/steps/UserSteps.java]: Is the annotation required?

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/steps/AuthenticationSteps.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+
+/**
+ * Class responsible for managing authentication steps.
+ * <p>
+ * Created by rsanchez on 23/02/18.
+ */
+@Component
```
Comment (crepache) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/steps/AuthenticationSteps.java:35]: Is the annotation required?

**Review by crepache — CHANGES_REQUESTED** [review]

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java [hunk 4]
```diff
@@ -102,6 +102,8 @@ public void bootstrap() {
   }
 
   private void prepareEnvironment() {
+    testContext.setCertsDir(System.getProperty("java.io.tmpdir"));
```
Comment (crepache) [inline helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/HelpDeskBotStories.java]: Add a constant on the class to the location of the certificates.
Private static String CERTS_DIR = "/certs";

and Change the location of certificates to System.getProperty("java.io.tmpdir") + CERTS_DIR

**Review by crepache — CHANGES_REQUESTED** [review]

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/steps/AuthenticationSteps.java (hunk 3 above)
Comment (robson-sanchez) [reply helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/steps/AuthenticationSteps.java:35]: Yes, it's required to support dependency injection of HelpDeskConfig class and then get the YAML settings.

File: helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/steps/TicketSteps.java (hunk 1 above)
Comment (robson-sanchez) [reply helpdesk-bot/src/test/java/org/symphonyoss/symphony/bots/helpdesk/bot/it/steps/TicketSteps.java:19]: Yes, it's required to support dependency injection of application Symphony client and then get the bot user info.

**Review by crepache — APPROVED** [review]
LGTM

**Pull request merged**